test/plugins/test-trace-hapi.js

```diff
@@ -308,6 +308,41 @@ describe('hapi', function() {
           });
         });
       });
+
+      it('should end spans when client aborts request', function(done) {
```

Comment:
The middleware runs to completion as far as I can tell. Yes, I can (and should) add those tests.